### ISSUE TITLE
vector binary search using recursion

### ISSUE THREAD

**Issue by daymos** [codingforeveryone/js#25]
Problem: find if an element x is present in a vector using binary search principle. The function must be written using recursion.

I think this is a classic problem, so if you look around you will find the solution already made. So I try to give here the basic explanation to solve it without having to look around too much. Probably you know already this topic, but I´ll write a couple of lines anyway.

Binary search is efficient, because it never uses more the log_2(N) attempt to find an element in a vector of N elements.

You need the data to be ordered. You cannot assume the data will be ordered. You can assume it will be an array of integers.

The principle is that you divide the data in 2 parts, and check if the number you are searching for, is bigger or smaller then the value you have on the division. If not you repeat with one of the 2 subset of data and go on until you find the value.

**Comment by patrickwalkowicz** [comment]
@daymos thanks for posting this problem!

Can we assume that the data passed to this function will be already sorted or do we need to handle unsorted data as well?

**Comment by daymos** [comment]
Better to check if it´s sorted. I´ll edit the description.

**Comment by patrickwalkowicz** [comment]
Thanks! Just making sure. If you don't mind, please also add info on if we should expect an array, object or multiple arguments to be passed as data?

**Comment by daymos** [comment]
Actually I just read your comment on the other discussion about making this repo more about learning github rather then JS, and I agree very much with it. In light of it, is probably better to keep it simple. (Although I dont know what passing double arguments mean and would be curious to see what it is..)

**Comment by patrickwalkowicz** [comment]
By multiple arguments I meant a function that could handle data passed as any number of arguments - for example it could be just 2 arguments or it could be 10 (depending on how the function is called). In short, it should take any number of arguments. 
So in the function you'd have to use a method that would allow you to access the arguments and learn something about JS that way. It doesn't seem like an elegant or efficient way of passing input but could be good for learning purposes :smile:

**Comment by daymos** [comment]
oh, ok. That is clear, but no in this case only one vector. 
I though you mean something else. 
I have seen on codewars something like this:   myfunction(1)(2)
And still have to get my head around it.

**Comment by JMurphyWeb** [comment]
Hey @daymos check out my findsum function in this repository which @patrickwalkowicz has solved for a very simplified version of the problem you are talking about there.

@daymos it was not my intention to ONLY have simple problems on here.  If you want to set a challenge then by all means you should go ahead.  My main worry is that people are put off adding exercises to the repo because they feel like they HAVE to make a challenging one. Both are useful so long as there are always problems to be solved!

**Comment by daymos** [comment]
Hi @JMurphyWeb, yes I am still thinking about this question, hard challenges vs easier on this repo. 
I think in essence is good to have the simple ones to practice github. But I noticed a difference between solving a challenge here and on codewars. I think I put more effort here. What about you guys? 
It may just be the human factor. Or, I think maybe people post here challenges they really find interesting. For example in this week I have decided to delve a bit deeper into recursive function, because I find the topic really interesting. And I like to work on topic that other people are passionate about.  So that inspired me to post this exercise. But it´s hard that people will be passionate about very simple problems :)

I will add example input this afternnoon. But consider, the function must take 2 argument. One is a vector of number (integers), and the other is a number to be found (integer). and the output is a simple true of false, if he found the number into the array.

I´ll look at your excercise later today.
